src/credentials/GenerateChipX509Cert.cpp

```diff
@@ -367,6 +365,61 @@ CHIP_ERROR EncodeTBSCert(const X509CertRequestParams & requestParams, const Cryp
     return err;
 }
 
+} // namespace
+
+CHIP_ERROR EncodeNetworkIdentityTBSCert(const P256PublicKey & pubkey, ASN1Writer & writer)
```

Comment:
Suggest refactoring EncodeTBSCert to use an anonymous `EncodeTBSCert(const X509CertRequestParams &requestParams, const Crypto::P256PublicKey * subjectPubkey,  const Crypto::P256PublicKey * issuerPubkey, ASN1Writer & writer)`

Where nullptr subjectPubKey/issuerPubKey allows conditionally omitting the SKI/AKI.

Then the method here would be a simpler call of common logic, rather than a fresh inline expansion here.

Reply:
I've factored out common stuff like encoding the extensions, however the top-level functions have slightly different assumptions: unlike the other certs the current function generates TBSCerts for, the PDC certs have a fully rigid structure that's mandated by the spec and allows them to be round-tripped through the compact format, whereas the other certificates we generate don't. It would be relatively easy to accidentally break this if additions or changes are made later to a common function.